### DISCUSSION TITLE
Grafana Mimir's PVC is full

### DISCUSSION THREAD

**Discussion by felipefrx** [grafana/mimir#8917] [Help and support] [answered]
Grafana mimir is sending metrics to s3. The ingesters have volumes enabled. But today I had a problem, the volumes filled up. Is there any parameter regarding the retention period of local storage?

OBS: My volume is 50gb

**Comment by pstibrany** [comment, marked as answer]
There is `-blocks-storage.tsdb.retention-period` option. It defaults to 13h. 

> TSDB blocks retention in the ingester before a block is removed. If shipping is enabled, the retention will be relative to the time when the block was uploaded to storage. If shipping is disabled then its relative to the creation time of the block. This should be larger than the -blocks-storage.tsdb.block-ranges-period, -querier.query-store-after and large enough to give store-gateways and queriers enough time to discover newly uploaded blocks.

Note that other options mentioned in the description have default values based on those 13h. If you change `-blocks-storage.tsdb.retention-period`, you will need to adjust those other options too.
